src/java/org/apache/cassandra/io/sstable/format/SSTableFormat.java

```diff
@@ -59,7 +60,18 @@ public interface SSTableFormat<R extends SSTableReader, W extends SSTableWriter>
      */
     Set<Component> allComponents();
 
-    Set<Component> streamingComponents();
+    /**
+     * Returns the components that should be streamed to other nodes on repair / rebuild.
+     * This includes only the core SSTable components produced by this format.
+     * Custom components registered by e.g. secondary indexes are not included.
+     * Use {@link SSTableReader#getStreamingComponents()} for the list of all components including the custom ones.
+     */
+    default Set<Component> streamingComponents()
```

Comment:
Once all `Component` instances have their "streamability" embedded, we don't need this method at all, right? i.e. `SSTableZeroCopyWriter` will no longer need special casing?

Reply:
Looks like we need it for testing. I'll think of a way to get rid of it completely.